components/ws-manager-api/go/crd/v1/workspace_types.go

```diff
@@ -180,6 +185,99 @@ const (
 	WorkspaceConditionBackupFailure WorkspaceCondition = "BackupFailure"
 )
 
+func NewWorkspaceConditionDeployed() metav1.Condition {
+	return metav1.Condition{
+		Type:               string(WorkspaceConditionDeployed),
+		LastTransitionTime: metav1.Now(),
+		Status:             metav1.ConditionTrue,
+	}
+}
+
+func NewWorkspaceConditionFailed(message string) metav1.Condition {
+	return metav1.Condition{
+		Type:               string(WorkspaceConditionFailed),
+		LastTransitionTime: metav1.Now(),
+		Status:             metav1.ConditionTrue,
+		Message:            message,
+	}
+}
+
+func NewWorkspaceConditionTimeout(message string) metav1.Condition {
+	return metav1.Condition{
+		Type:               string(WorkspaceConditionTimeout),
+		LastTransitionTime: metav1.Now(),
+		Status:             metav1.ConditionTrue,
+		Reason:             "TimedOut",
+		Message:            message,
+	}
+}
+
+func NewWorkspaceConditionFirstUserActivity() metav1.Condition {
+	return metav1.Condition{
+		Type:               string(WorkspaceConditionFirstUserActivity),
+		LastTransitionTime: metav1.Now(),
+		Status:             metav1.ConditionTrue,
+		Reason:             "MarkActiveRequest",
```

Comment:
maybe we should change the `Reason` now, as it might be called from other places than the MarkActive request handler